src/Controller/Admin/OrderOverview.php

```diff
@@ -140,4 +143,47 @@ public function refundpayment()
             );
         }
     }
+
+    public function makecharge()
+    {
+        $oOrder = oxNew(Order::class);
+        /** @var float $captureAmount */
+        $captureAmount = Registry::getRequest()->getRequestParameter("captureAmount");
+        $amazonConfig = oxNew(Config::class);
+        $currencyCode = $order->oxorder__oxcurrency->rawValue ?? $amazonConfig->getPresentmentCurrency();
+        $orderLoaded = $oOrder->load($this->getEditObjectId());
+        /** @var string $paymentType */
+        $paymentType = $oOrder->getFieldData('oxpaymenttype');
+
+        if (
+            $orderLoaded &&
+            Constants::isAmazonPayment($paymentType) &&
+            $oOrder->getId() !== null
+        ) {
+            $chargeId = '';
+
+            $repository = oxNew(LogRepository::class);
+            $logMessages = $repository->findLogMessageForOrderId($this->getEditObjectId());
+            if (!empty($logMessages)) {
+                foreach ($logMessages as $logMessage) {
+                    $logsWithChargePermission = $repository->findLogMessageForChargePermissionId(
+                        $logMessage['OSC_AMAZON_CHARGE_PERMISSION_ID']
+                    );
+                    foreach ($logsWithChargePermission as $logWithChargePermission) {
+                        if ($logWithChargePermission['OSC_AMAZON_RESPONSE_MSG'] === 'Captured') {
+                            return '-1';
```

Comment:
is it correct to return here instead of break?

Reply:
mario will fix the code smells